lib/puppet/provider/package/pkgng.rb

```diff
@@ -61,7 +61,7 @@ def self.instances
   end
 
   def self.prefetch(resources)
-    packages = instances
+    packages = instances || []
```

Comment:
Thanks for the review @Iristyle.  Ah, you're correct, I've not modified the instance to return an empty package list.  I can reword as well, or give more backstory in the commit/pr/ticket.  This didn't used to be a problem until a relatively recent version of pkg(8) began exiting non-zero for some cases.  This only currently comes up when there are no repositories, or at least that is the use case I'm targeting.  The solution as it stands now (wording aside) should mean that calls to `pkg` exiting non-zero causing a failure in the `instances` call ends up with an empty package list rather than bad provider objects.  I think I leaned that way out of habit for other types/providers I work on where the results of instances aren't as clean as they should be.

Do you think modifying line 58 of the provider here return an empty array rather than nil is the right move then?  It seems like that would be the other simple option for this change.